mathics/builtin/makeboxes.py

```diff
@@ -394,6 +394,10 @@ def apply_general(self, expr, f, evaluation):
         """MakeBoxes[expr_,
         f:TraditionalForm|StandardForm|OutputForm|InputForm|FullForm]"""
         if isinstance(expr, BoxElementMixin):
+            # If we are inside a DisplayForm block,
+            # BoxElementMixin are not processed.
+            if evaluation.in_display_form:
```

Comment:
Can this be handled by some sort of `HoldForm` attribute?

Reply:
I don't think so. In the first place, I guess that in WMA ,  it does not have that attribute for a reason. But, I also tried to avoid the new attribute, and I could not figure out how to do that without changing more parts of the code.

Reply:
Ok - I'd like to check on this. Doing this kind of thing feels weird.

Reply:
Thinking about it again, one alternative would be to set a symbol in a private context, and then instead of looking at that property, look at the value of the symbol. This could have the advantage to make easy to replace the Python code that implements MakeBoxes by WL code. But in the end, it would be the same idea.

Reply:
To start out, I believe we should be following the way that can be written in WL would do that. This keeps us from inventing mechanisms, and flags and so on. Also, since WL follows a certain logic, makes it more likely we will go with the natural flow here. 

If later we need to get things faster then starting with the logical flow is better than some guessed version.

Reply:
As I think I mentioned on the other PR, `TagBox` is used by `DisplayForm` and that is probably a mechanism by which we get special behavior. 

See https://reference.wolfram.com/language/ref/TraditionalForm.html